### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by kshitijthatte**
- created auth context
- added login page
- added signup page

### CODE REVIEW

**Comment by MiheerTamkhane** [comment]
image

Error 500 is occuring?

**Comment by kshitijthatte** [comment]
> Error 500 is occuring?

@MiheerTamkhane Can you please have a look once again? I forgot to set up the environment variables.

**Comment by MiheerTamkhane** [comment]
Yup its working fine rn.

**Review by MiheerTamkhane — APPROVED** [review]

File: src/components/Navbar.js [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
+            </NavLink>
+            <button
+              className="btn btn-primary btn-sm btn-icon"
+              onClick={() => signoutHandler(setAuth, navigate)}
+            >
+              <span className="material-icons"> logout </span>
+            </button>
+          </>
+        )}
+        {/* <a className="btn btn-primary btn-sm btn-icon" href="/">
```
Comment (sakshamAK) [inline src/components/Navbar.js:39]: you can remove this comment if not needed.

File: src/pages/LoginPage/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      });
+      localStorage.setItem("AUTH_TOKEN", response.data.encodedToken);
+      localStorage.setItem("WB_USER", JSON.stringify(response.data.foundUser));
+      setAuth(a => ({
+        ...a,
+        isAuthenticated: true,
+        user: response.data.foundUser,
+        token: response.data.encodedToken,
+      }));
+      navigate(`/`);
```
Comment (sakshamAK) [inline src/pages/LoginPage/index.js:36]: Just a suggestion, you can add an empty line after lines 22, 27, 29, 35. This makes it cleaner and easier to identify where the code block ends. (according to my opinion)

File: src/pages/LoginPage/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              setInputValues(v => ({ ...v, email: e.target.value }))
+            }
+          />
+          <input
+            type="password"
+            className="input"
+            placeholder="Password"
+            value={inputValues.password}
+            onChange={e =>
+              setInputValues(v => ({ ...v, password: e.target.value }))
```
Comment (sakshamAK) [inline src/pages/LoginPage/index.js:63]: v doesn't seem to justify its meaning here, you can use a different variable name.

**Review by sakshamAK — APPROVED** [review]
your code is clean and easy to read and the folder structure is good too. I liked how you named one of the folders as services, it really justifies its use case.

File: src/components/Navbar.js (hunk 1 above)
Comment (kshitijthatte) [reply src/components/Navbar.js:39]: Thanks for pointing that out, will change it to a `Link` after implementing Private routes

File: src/pages/LoginPage/index.js (hunk 2 above)
Comment (kshitijthatte) [reply src/pages/LoginPage/index.js:36]: Sure, will do that.

File: src/pages/LoginPage/index.js (hunk 3 above)
Comment (kshitijthatte) [reply src/pages/LoginPage/index.js:63]: It's a common convention to name the pending state argument for the first letter of the state variable name, like here `v` for  `values`. I read this in the react docs.

**Pull request merged**